client.go

```diff
@@ -159,8 +159,12 @@ func (c *APIClient) Patch(url string, payload interface{}) (*http.Response, erro
 
 // Delete performs a Delete request against the Redfish service.
 func (c *APIClient) Delete(url string) error {
-	_, err := c.runRequest("DELETE", url, nil)
-	return err
+	resp, err := c.runRequest("DELETE", url, nil)
+	resp.Body.Close()
```

Comment:
I wonder if it would be better to handle this in a more common way. Have you tried adding something like this around line 242?

if resp.Body != nil {
    defer resp.Body.close()
}

Reply:
I would agree, however closing the body here leads to reads on the closed body due to downstream consumption of the returned resp object, so all usage of runRequest would need to be refactored or the body copied and then closed prior to return. What do you feel is the best path?

Reply:
I was thinking we could extract what's needed out of the response body, then use that as the output to make sure this is never left unclosed by other consumers of the call. But that may require more thought.

This at least makes this part better, so no reason to hold things up here. Thanks!